app/gui/src/model/suggestion_database.rs

```diff
@@ -596,4 +780,159 @@ mod test {
         assert_eq!(db.lookup(3).unwrap().arguments[2].name, "NewArg");
         assert_eq!(db.version.get(), 8);
     }
+
+    #[test]
+    fn lookup_by_fully_qualified_path() {
```

Comment:
These tests functions are huge. Split them down to smaller tests or add comments of what is happening on every step there. A test function should be easy to understand of WHAT behavior it is exactly testing. After spending several minutes, I do not know what this code is testing.

Reply:
I think the "lookup" and "modifying" should be separate tests.

Reply:
I split the tests into smaller ones. I didn't add many comments, as I hope the tests are more readable now and thus don't need them. But if they're still not clear enough, please let me know.